apps/app/src/storage/localstorage/AsyncStorage.ts

```diff
@@ -0,0 +1,20 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+export const storeData = async <T>(value: T, key: string): Promise<boolean> => {
```

Comment:
This can only ever return `Promise<true>`, because you either return true or throw an error. The `Promise<boolean>` creates the illusion that `false` could be returned.

I'd recommend using `Promise<void>` in this case and not returning `true`. The return value is actually not interesting here, you just want to know if you didn't get an error. That is already handled by default because you are throwing an `Error` when something goes wrong, so wherever you call `storeData()` you can surround it with a `try-catch` to check that.

Reply:
[try catch setStorage](https://github.com/gabrielakuhn/questline-galaxy/commit/c1741551031183c87762471382e33557c7fd82d1)